### PR TITLE
Cleanup rules.mk for 32U4 keyboards, H

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

H, and handwired boards.
The filesets should be a bit smaller from this point on.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/magicforce68/rules.mk [hunk 1]
```diff
@@ -1,52 +1,15 @@
...
-
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = halfkay
```
Comment (zvecr) [inline keyboards/handwired/magicforce68/rules.mk]: > is handwired with an Arduino Micro
> To flash the firmware onto the microcontroller, run make handwired/magicforce68:default:avrdude
not halfkay but not caterina, i suggest reverting back to `BOOTLOADER_SIZE`

File: keyboards/handwired/gamenum/rules.mk [hunk 2]
```diff
@@ -1,52 +1,15 @@
...
-
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = halfkay
```
Comment (zvecr) [inline keyboards/handwired/gamenum/rules.mk]: > Hardware Supported: GameNum, Pro Micro

```suggestion
BOOTLOADER = caterina
```

File: keyboards/handwired/magicforce68/rules.mk (hunk 1 above)
Comment (fauxpark) [reply keyboards/handwired/magicforce68/rules.mk]: Arduino Micro is still Caterina: https://arduino.stackexchange.com/questions/36147/what-bootloader-does-the-arduino-micro-come-with

**Comment by mtei** [comment]
Can you not change rules.mk under `keyboards/helix` ? Because the same change is already planned in #6952. So this PR will conflict.

**Comment by fauxpark** [comment]
There are some changes here that that PR doesn't do (removing F_CPU, ARCH etc). This is what merge conflict resolution is for ;)

**Comment by mtei** [comment]
In #6952, if I remove the following, will it be solved?

```
F_CPU = 16000000
```
```
ARCH = AVR8
```
```
F_USB = $(F_CPU)
```
```
# Interrupt driven control endpoint task(+60)
OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
```

**Comment by fauxpark** [comment]
Yep. I made the suggestion on the PR.

**Comment by fauxpark** [comment]
Fixed

**Pull request merged**